languages/C.groovy

```diff
@@ -0,0 +1,379 @@
+                }
+            }
+        }
+    }
+
+    // clean up passed DD statements
+    job.stop()
+}
+
+// end script
+
+
+//********************************************************************
+//* Method definitions
+//********************************************************************
+
+/*
+ * createCParms - Builds up the c compiler parameter list from build and file properties
+ */
+def createCParms(String buildFile, LogicalFile logicalFile) {
+    def parms = props.getFileProperty('cc_compileParms', buildFile) ?: ""
+    def cics = props.getFileProperty('cc_compileCICSParms', buildFile) ?: ""
+    def sql = props.getFileProperty('cc_compileSQLParms', buildFile) ?: ""
+    def errPrefixOptions = props.getFileProperty('cc_compileErrorPrefixParms', buildFile) ?: ""
+    def compileDebugParms = props.getFileProperty('cc_compileDebugParms', buildFile)
+
+    if (buildUtils.isCICS(logicalFile))
+        parms = "$parms,$cics"
+
+    if (buildUtils.isSQL(logicalFile))
+        parms = "$parms,$sql"
+
+    if (props.errPrefix)
+        parms = "$parms,$errPrefixOptions"
+
+    // add debug options
+    if (props.debug)  {
+        parms = "$parms,$compileDebugParms"
+    }
+
+    if (parms.startsWith(','))
+        parms = parms.drop(1)
+
+    if (props.verbose) println "c compiler parms for $buildFile = $parms"
+    return parms
+}
+
+/*
+ * createCompileCommand - creates a MVSExec command for compiling the c program (buildFile)
+ */
+def createCompileCommand(String buildFile, LogicalFile logicalFile, String member, File logFile) {
+    String parms = createCParms(buildFile, logicalFile)
+    String compiler = props.getFileProperty('cc_compiler', buildFile)
+
+    // define the MVSExec command to compile the program
+    MVSExec compile = new MVSExec().file(buildFile).pgm(compiler).parm(parms)
+
+    // add DD statements to the compile command
+    if (isZUnitTestCase){
+        compile.dd(new DDStatement().name("SYSIN").dsn("${props.cc_testcase_srcPDS}($member)").options('shr').report(true))
+    }
+    else
+    {
+        compile.dd(new DDStatement().name("SYSIN").dsn("${props.cc_srcPDS}($member)").options('shr').report(true))
+    }
+
+    compile.dd(new DDStatement().name("SYSOUT").options(props.cc_tempOptions))
+    compile.dd(new DDStatement().name("SYSPRINT").options(props.cc_tempOptions))
+
+    compile.dd(new DDStatement().name("SYSMDECK").options(props.cc_tempOptions))
+    (1..17).toList().each { num ->
+        compile.dd(new DDStatement().name("SYSUT$num").options(props.cc_tempOptions))
+    }
+
+    // Write SYSLIN to temporary dataset if performing link edit or to physical dataset
+    String doLinkEdit = props.getFileProperty('cc_linkEdit', buildFile)
+    String linkEditStream = props.getFileProperty('cc_linkEditStream', buildFile)
+    String linkDebugExit = props.getFileProperty('cc_linkDebugExit', buildFile)
+
+    if (props.debug && linkDebugExit && doLinkEdit.toBoolean()){
```

Comment:
In a previous zAppBuild release, we have [simplified the management of the OBJs](https://github.com/IBM/dbb-zappbuild/blob/a719842f8b9738133e375eaade8497289c30fc05/languages/Cobol.groovy#L204-L205), and always allocate a dataset, that is referenced in the Link step instead of using a temp dataset. The contribution was made in #245 

It is preferred to follow this structure for C/C++ as well.